tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -1658,7 +1658,14 @@ class PlaceholderExpansionScanner {
         auto SR = E->getSourceRange();
         if (SR.isValid() && SM.rangeContainsTokenLoc(SR, TargetLoc) &&
             !checkCallExpr(E) && !EnclosingCallAndArg.first) {
-          OuterExpr = E;
+          if (!isa<TryExpr>(E) && !isa<AwaitExpr>(E) &&
```

Comment:
`getSemanticsProvidingExpr` covers all `IdentityExpr`. FWIW you probably want to reset `OuterExpr` in these cases as well, ie. even with this change something like the following would fail:
```
bar(try foo(x: <#T##() -> Void#>))
```

And I haven't thought too hard about other cases, but TBH it doesn't really matter, we'll replace this with the swift-syntax one soon :).

Reply:
I did not want to use `getSemanticsProvidingExpr` because IIRC that includes e.g. `.self`. And IIUC we don’t want to expand

```swift
foo(<#placeholder#>).self
```

to

```swift
foo { <#code#> }.self
```

Or at least that’s what I understood the purpose of `OuterExpr` is.

Reply:
I'd be fine with `.self`. I'm not really sure I agree with what we're trying to prevent here anyway. Ie. the comment says that `outer(inner(<#closure#>))` shouldn't expand to trailing closure for `inner`. But that also seems fine to me 🤷. Certainly the swift-syntax replacement doesn't currently care about that. So something we can talk about later.

Like I said, I'm perfectly happy with what you have now.